metaflow/plugins/kfp/kfp.py

```diff
@@ -1072,13 +1070,15 @@ def build_kfp_dag(
                 )
                 KubeflowPipelines._set_minimal_container_resources(workflow_uid_op)
 
-            s3_sensor_op: ContainerOp = None
-            s3_sensor_deco = self.flow._flow_decorators.get("s3_sensor")
-            if s3_sensor_deco:
-                s3_sensor_op = self._create_s3_sensor_op(
-                    s3_sensor_deco=s3_sensor_deco,
-                    flow_parameters_json=flow_parameters_json,
-                )
+            def create_s3_sensor_op():
+                s3_sensor_op: ContainerOp = None
+                s3_sensor_deco = self.flow._flow_decorators.get("s3_sensor")
+                if s3_sensor_deco:
+                    s3_sensor_op = self._create_s3_sensor_op(
+                        s3_sensor_deco=s3_sensor_deco,
+                        flow_parameters_json=flow_parameters_json,
+                    )
+                return s3_sensor_op
```

Comment:
style: it's easier to read if it were like this:

```python
                s3_sensor_deco = self.flow._flow_decorators.get("s3_sensor")
                if s3_sensor_deco:
                    return self._create_s3_sensor_op(
                        s3_sensor_deco=s3_sensor_deco,
                        flow_parameters_json=flow_parameters_json,
                    )
                else:
                    return None
```

Reply:
The control path and structure is easier to reason, and you get to remove a whole variable.